csrc/python_frontend/segmentation.cpp

```diff
@@ -35,31 +35,52 @@ int64_t SegmentationState::setupSegmentation(
   IrCloner original_to_cloned_map =
       Fusion::copy(fusion, cloned_original_fusion_.get());
 
+  // Step 2) Given the map_presched_value_to_original_python_index AND the
```

Comment:
Originally, this step mapped from `prescheduled_fusion` to `cloned_concretized_fusion`, which causes issue with DEBUG build.

The fix is to separate the step into a transform from `prescheduled_fusion` to `cloned_fusion`, apply concretization, then transform from `cloned_fusion` to `cloned_concretized_fusion`.

Reply:
QQ: what's the issue you see on DEBUG build? Are we accessing dangling pointers or something?

Reply:
i.e. if there's some check that we only enabled in DEBUG build, we might want to also include that in our CI tests.

Reply:
@xwang233 I believe we run the DEBUG build in the CI tests. The CI found the failure for me.

It found an issue through `as<Val>()` where dynamic cast is used.

Reply:
we don't run DEBUG build but we run `--explicit-error-check` build and all tests are executed on builds with explicit error check

Reply:
nvm, we are all good then. Thanks for bringing this up @rdspring1